### PR TITLE
Add formatter workflow

### LINKED ISSUE #24: Deploy a code formatter for the bindings
Currently we do not have a formatter which is governing the code styling for the bindings. We should investigate which is the most standard format to write Julia code and accordingly standardize the code and also if possible deploy a formatting action which will check if code added in the future is properly formatted or not.

One option could be the [JuliaFormatter](https://juliapackages.com/p/juliaformatter) which is also [used in VS Code](https://www.julia-vscode.org/docs/dev/userguide/formatter/).

### PR DESCRIPTION
**Pull request by erikscheurer**
closes #24 

This PR adds a workflow from [julia-actions](https://github.com/julia-actions/julia-format/blob/master/workflows/format_check.yml), that checks if the julia code is formatted correctly.
The current code is also formatted using [JuliaFormatter.jl](https://github.com/domluna/JuliaFormatter.jl)

### CODE REVIEW

**Comment by erikscheurer** [comment]
The README update from #30 is accidentally also included in this PR, this has nothing to do with formatting.

File: .github/workflows/format.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+          julia -e '
+          out = Cmd(`git diff --name-only`) |> read |> String
+          if out == ""
+              exit(0)
+          else
+              @error "Some files have not been formatted !!!"
+              write(stdout, out)
+              exit(1)
+          end'
+
```
Comment (IshaanDesai) [inline .github/workflows/format.yml:43]: Newline necessary here?

**Review by IshaanDesai — APPROVED** [review]
Looks good :+1: lets revert the README changes which have been introduced unintentionally before merging this.

File: .github/workflows/format.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
+name: format-check
+
+on:
+  push:
+    branches:
+      - 'main'
+      - 'release-'
+      - 'develop'
```
Comment (IshaanDesai) [inline .github/workflows/format.yml:8]: Why dont we run this on all branches? Just noticed this, sorry.. lets have this Action run on all branches for every push

**Pull request merged**